File: 728_self_dividing_numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> res = new ArrayList<>();
        for (int curr = left; curr <= right; curr++){
            if (isSelfDivided(curr)) {
                res.add(curr);
            }
        }
        return res;
    }

    public boolean isSelfDivided(int num) {
        char[] digits = String.valueOf(num).toCharArray();
        for (char c: digits) {
            if (c == '0') return false;

            int digit = c - '0';
            if (num % digit != 0) return false;
        }
        return true;
    }
}

File: python/728_self_dividing_numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        while left <= right:
            if 0 < left < 10:
                res.append(left)
                left += 1
                continue

            if left % 10 == 0:
                left += 1
                continue

            isSelfDividing = True
            for c in str(left):
                if int(c) == 0 or left % int(c) != 0:
                    isSelfDividing = False
                    break
            if isSelfDividing:
                res.append(left)

            left += 1
        return res
